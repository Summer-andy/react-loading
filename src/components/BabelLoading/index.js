import React from 'react';
import styled, { keyframes } from 'styled-components';
import { commonStyle } from '../utils/style';

const animation = keyframes`
	0% {
		-webkit-transform: scale(0.1);
		opacity: 1;
	}
	50% {
		-webkit-transform: scale(1);
		opacity: 0.5;
	}
	100% {
		-webkit-transform: scale(1.5);
		opacity: 0.1;
	}
`

const Container = styled.div`
  height: 50px;
  width: 50px;
  border-radius:25px;
  background:  ${props => props.color || '#00adb5'};
  animation: ${animation}  ${props => props.speed || 2}s linear infinite;
`

const BabelLoading = ({ style = commonStyle, speed, color }) => {
  return (
    <Container {...{style, speed, color }} />
  );
};

export default BabelLoading;
