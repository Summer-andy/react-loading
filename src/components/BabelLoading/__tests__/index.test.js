import React from 'react'
import renderer from 'react-test-renderer'
import BabelLoading from '../index'

it('renders correctly', () => {
  const tree = renderer.create(<BabelLoading speed={1} color="red"  />).toJSON()
  expect(tree).toMatchSnapshot()
})
