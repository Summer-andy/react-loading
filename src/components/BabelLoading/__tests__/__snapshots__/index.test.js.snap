// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa ieeOUE"
  color="red"
  size="default"
  speed={1}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "margin": "auto",
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
/>
`;
